File: JS/MakeTheStringGreat.js
/**
 * @version JAVASCRIPT ECMAScript 6
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */

/**
 * @param {string} s
 * @return {string}
 */
function makeGood(s) {
    const stack = [];
    for (let i = 0; i < s.length; i++) {
        if (stack.length !== 0 && (s[i].charCodeAt(0) === stack[stack.length - 1].charCodeAt(0) + 32 || s[i].charCodeAt(0) === stack[stack.length - 1].charCodeAt(0) - 32)) {
            stack.pop();
            continue;
        }
        stack.push(s[i]);
    }
    return stack.join('');
}
export { makeGood }

File: GO/MakeTheStringGreat.go
/**
 * @version GO 1.22.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

func makeGood(s string) string {
	stack := []byte{}
	for i := 0; i <= len(s)-1; i++ {
		if len(stack) != 0 && (s[i] == stack[len(stack)-1]+32 || s[i] == stack[len(stack)-1]-32) {
			stack = stack[:len(stack)-1]
			continue
		}
		stack = append(stack, s[i])
	}
	return string(stack)
}
